src/relation/schema.rs

```diff
@@ -98,10 +101,124 @@ impl Schema {
         self.fields.iter()
     }
 
+    fn update(self, column: &Identifier, datatype: DataType) -> Schema {
+        let name = column.head().unwrap();
+        let new_fields: Vec<Field> = self
+            .into_iter()
+            .map(|f| {
+                if f.name() == name {
+                    Field::from((name.to_string(), datatype.clone()))
+                } else {
+                    f
+                }
+            })
+            .collect();
+        Schema::new(new_fields)
+    }
+
     /// Returns a new `Schema` where the `fields` of this `Schema`
     /// has been filtered by predicate `Expr`
-    pub fn filter(&self, predicate: &Expr) -> Self {
-        Self::new(self.iter().map(|f| f.filter(predicate)).collect())
+    pub fn filter(&self, predicate: &Expr) -> Result<Self> {
```

Comment:
A column in an expression is not sufficient to know which field is designated. You need a mapping from Identifiers to names.